### PR TITLE
S6: Integration with upstream

### PR DESCRIPTION
**Pull request by bryanlatten**
- [x] running php-fpm master process non-root
- [x] replaced supervisor with s6
- [x] swapped compiling apcu with packaged version
- [x] combined expensive layers (ugly but better shrink-wrapping)
- [x] save `>100MB` vs. 3.0 (some from the parent being smaller too :) )

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM behance/docker-nginx:3.0
+FROM behance/docker-nginx:4.0-edge
```
Comment (bryanlatten) [inline Dockerfile]: Following: https://github.com/behance/docker-nginx/wiki/Release-Milestones

**Comment by ingluisjimenez** [comment]
:+1:

File: Dockerfile [hunk 2]
```diff
@@ -59,16 +62,11 @@ RUN apt-get update && \
...
     echo 'extension=igbinary.so' > $CONF_PHPMODS/igbinary.ini && \
     php5enmod igbinary && \
-    printf "\n" | pecl install apcu-4.0.7 && \
-    echo 'extension=apcu.so' > $CONF_PHPMODS/apcu.ini && \
-    php5enmod apcu && \
     # Ensure development/compile libs are removed \
     curl -sS https://getcomposer.org/installer | php && \
     mv composer.phar /usr/local/bin/composer && \
-    ./clean.sh
+    /bin/bash /clean.sh
```
Comment (bryanlatten) [inline Dockerfile:68]: prep for `alpine` linux :)

File: docker-compose.yml [hunk 3]
```diff
@@ -11,5 +11,7 @@ web:
     SERVER_APP_NAME: docker-test
     REPLACE_NEWRELIC_APP: abcdefg
     REPLACE_NEWRELIC_LICENSE: hijklmno
+    S6_KILL_FINISH_MAXTIME: 1
+    S6_KILL_GRACETIME: 1
```
Comment (bryanlatten) [inline docker-compose.yml:15]: dev users will probably want these environment variables set in their compose file to make stack takedown really fast

File: container/root/etc/services-available/php-fpm/run [hunk 4]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/with-contenv bash
+
+exec sudo -u www-data php5-fpm -F -O 2>&1 | sed -u 's,.*: \"\(.*\)$,\1,'| sed -u 's,"$,,' 1>&1
```
Comment (mannytoledo) [inline container/root/etc/services-available/php-fpm/run]: Did you get this piping to work?

Comment (bryanlatten) [inline container/root/etc/services-available/php-fpm/run]: @mannytoledo the piping works, but the signal passing does not. if the container is being taken down, nginx will respond to the draining, which will close off the backends. in this case, we're working around the problem, but its not different than it is currently today

**Comment by mannytoledo** [comment]
Welp too late but :+1:

**Pull request merged**